### PR TITLE
[CALCITE-3834] Support AntiJoin in EnumerableMergeJoin

### PR DESCRIPTION
**Pull request by rubenada**
[[CALCITE-3834](https://issues.apache.org/jira/browse/CALCITE-3834)] Support AntiJoin in EnumerableMergeJoin

### CODE REVIEW

**Comment by rubenada** [comment]
@hsyuan the behavior of our JoinType ANTI is not like a NOT IN, but a NOT EXISTS.
Some time ago, I added a test case for this precise scenario in EnumerableCorrelateTest#antiJoinCorrelateWithNullValues (there is a comment in there explaining the situation).
Bear in mind that, at that time, ANTI was a CorrelateJoinType, but not a JoinType. Later, with the big Join refactoring, ANTI (together with SEMI) was added as JoinType, and CorrelateJoinType was deprecated (and later removed). So, we must see ANTI as a NOT EXISTS, whose null management is not as drastic as NOT IN.

**Comment by hsyuan** [comment]
@rubenada Thanks for the clarification. That make much sense now.

File: linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java [hunk 1]
```diff
@@ -3878,6 +3879,8 @@ public void remove() {
     private final Comparator<TKey> comparator;
     private boolean done;
     private Enumerator<TResult> results;
+    // only used for ANTI join: if right input is over, all remaining elements from left are results
+    private boolean remainingLeft;
```
[the remark this answers is not in the document]
Comment (rubenada) [reply linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java:3883]: Correct, it means the right input is over, but the left is not; hence all remaining tuples from the left relation are results for the anti-join

**Comment by rubenada** [comment]
@hsyuan gentle reminder. I would to have this one reviewed, so that it can get into the next release.

**Comment by hsyuan** [comment]
Sorry for my late action. Will do it tomorrow. :)

**Pull request merged**